### ISSUE TITLE
Presentation summary

### ISSUE THREAD

**Issue by ngraymon** [ngraymon/DNN-SE#12]
Note:

- All teammates should participate in the presentation done by their group and every group member has to get an equal chance to speak on a segment of the slides.
- Make sure all your group members are available to be present in the class and talk on that day before you sign up, because if you sign up to reserve a time and you end up needing to change your date you will have to convince another group to swap with you.

Format:
- You will present a slide deck in the class and talk over it. Group lead should send me the slides before noon of your presentation. Since creating exposure to applications of ML in physics is part of the objectives of the course, your slide deck will be made available to your classmates as part of the course material. The time and title of each presentation will also be posted after the sign-up round is over. Your presentation is 20 minutes followed by 10 minutes of Q&A. In the Q&A segment you address questions from your audience (me and your classmates) and I might have further feedback regarding your project plans.
- You should coordinate for all group members to have an equal opportunity to speak (~5min per person for groups of size 4 and ~4min per person for groups of size 5). Your team has to remain strictly within the 20-minute time limit. Consider doing internal dry-runs of your presentations in your group; consider what questions might come up; and, discuss how you'd address them.

Content:
- The main goal of the midterm presentation is to demonstrate that:
   1. your group has read the paper and understood it properly
   2. you have a clear idea of what you'd be doing in your course project.
 
Suggested Outline:

1. Problem statement. What is the problem tackled by your paper? Give an introduction to the scientific background assumed and/or discussed in the paper, and the computational challenge it intends to overcome using ML.
    - [ ] It probably makes the most sense for me to do this part. The problem is to solve the Schrodinger equation, i.e. has all the answers for any quantum mechanical system, pretty much what all of chemistry/physics is trying to do.

2. Methodology. How does the paper solve this problem using ML? Give a summary of the ML techniques, methods, learning paradigm, models, and architectures used by the paper and how the proposed solutions solve the problem stated in (1). Be prepared for questions regarding conventional state-of-the-art for the problem and why/if ML is a better technique to solving the problem.
    - [ ] It probably makes the most sense for William and JB to explain both the network and the loss function?

3. Results. What results did the authors achieve? Give a summary of the original numerical experiments done in the paper. Show the tables, figures, charts, etc. reported by the paper and explain what is reported/claimed in each figure on the assessed application of ML to solving the problem stated in (1).
    - [ ] Maybe Andrew should explain this part?

4. Your roadmap. What would you be doing in the coming month? Give a clear presentation of what parts of the paper you intend to focus on for your final project. Discuss further details on how you'd generate training/validation data; how you'd implement the experiment; what compute resources you have access to or you'd need access to; what reports do you expect to be able to generate/reproduce; what challenges you think you might be facing; how you'd break the upcoming month into reasonable milestones to ensure your project stays on track. You should also speak to project management considerations: how do you plan to divide the tasks between yourselves; what will the role of each group member be; who's task will depend on completion of other person's task; how do your resolve and optimize your dependencies; which milestones have a higher risk of incurring delays; and, what your plans are in case you face such delays.
   - [ ] Matthew could do this part?

Note:
- Sections (1) to (3) demonstrate your understanding of the paper. You should make sure you all understand the paper very very well and have done ample background reading on the topic and clarified your understanding. You should all be able to answer technical questions about the paper (not just the person presenting that piece). Section (4) is a very critical part of your presentation. You should work with your team to develop a reasonable set of milestones. Some additional guidelines follow.

**Comment by ngraymon** [comment]
Objectives:

### One
> You should provide a full citation of the paper on one of your intro slides and mention whether you are following an arXiv version or a journal version. Journal version is preferred because it might incorporate some feedback from journal reviewers, however many journals have page/space limits and therefore authors might have to shuffle their manuscript and/or refer the reader back to a different version (or supplementary material) for details. As a result, the arXiv version might appear to be a better version to follow. At any rate, you should mention all these considerations
- [ ] Provide a full citation of the paper on an intro slide and mention that we are following a journal version.
 
### Two
> It is possible that the paper has more results than you can feasibly reproduce in a month. In that case, consider selecting a subset of the results of the paper, or perhaps similar results but done on smaller scales (e.g. smaller system sizes, smaller dataset, smaller or different type of network, etc.). You should present these decisions as you talk about the paper results in Section (3). I might have other suggestions or feedback that will affect your decision. So be prepared for possible changes from what you propose. If I still had further feedback about the design and choice of numerical experiments that did not fit the 10 min Q&A round, I'd provide them after the presentation.
- [ ] Select what results we think we can feasibly reproduce in a month ( i think we should simply start with Helium; using Hydrogen as a MVP for implementation)
 
### Three
> If the dataset has to be generated by you, consider allocating appropriate amount of time for it. Discuss how you'd generate the data and how long it might take you to do so. If the dataset is to be accessed from a different resource (e.g. through a website, by contacting authors, etc.) try to have it done before your presentation to remove this uncertainty from your timeline and roadmap.
- [x] The FNN does not use an external data set and is instead an ab-initio method and learns through the minimization of the expectation value of the energy <E>

### Four
> You should pay attention to the compute resources used by the paper and mention that information in Sections (2) and/or (3). If you already have the compute resources you need for your project mention that. If you are hoping I can make compute resources available to you mention that (I plan to get some time on Google Cloud Platform with a couple of options: high-cpu + no gpu, or moderate-cpu + gpu but I might change the specs of the machines depending on your project needs and if it is feasible for my budget).
Try to make an estimate of how long each experiment might take. Usually some indications can be found inside the paper. You might be able to do some preliminary tests and extrapolate the training times from them. If you felt there is too much uncertainty around this you should contact the paper authors and try to collect further information.

- [ ] I have access to the gpus on graham, as well as some private server gpus. Do any of you guys have personal Nvidia gpus that we could use? Per their setup we simply turn to page 15:

> For the FermiNet, all code was implemented in Tensor-
> Flow 1 built with CUDA 9. All experiments for systems
> with less than 20 electrons were run in parallel on 8 V100
> GPUs, while 16 GPUs were used for larger systems. With
> a smaller batch size we were able to train on a single GPU
> but convergence was significantly and disproportionately
> slower. For instance, ethene converged after just 2 days of
> training with 8 GPUs, while several weeks were required
> on a single GPU. Bicyclobutane, with 30 electrons, took
> roughly 1 month on 16 GPUs to train.


### Five
> If the code of the project is available on github or elsewhere, you must mention this in your presentation and explain how you plan to interact with that code. Will you completely ignore that code? Will you be using any parts of it to guide your development? Are you planning to develop code from scratch and/or to use alternative/improved software design compared to the existing code? How would differences between the results you'd be reproducing and what is produced in the paper play a part here? Give a justification to each such consideration.

- [ ] We need to mention that the code is available on github. However! They implemented the FNN in TensorFlow and therefore we can't directly use their code as the course expects us to use PyTorch for all our work. We will approach the code as follows:
    1. Draft up pseudo code/an outline of how we expect the NN to work and how the data will be processed.
    2. Compare our draft to the TensorFlow code; look for mistakes/inconsistencies, revise the draft
    3. Implement a MVP in PyTorch
    4. Test this MVP using the Hydrogen atom as a benchmark atom to make sure it is working and can produce results
    5. If successful move onto Helium gas as our target for replication



### Six 

> In Section (4) you should break down your project into smaller milestones with their designated timeline and discuss the workload for each milestone. You should include some sort of a Gantt chart and talk over it. I don't mind what template you use for the chart, or how simple/detailed/sophisticated the chart is. But it's a useful visualization for your timeline and milestones.

### Seven
